doc/static-route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,452 @@
+
+|    Keyword    | Description |
+|:-----------------:|:-----------:|
+| vrf | The static routes will be configured in the specified VRF. If not specified, default VRF will be used.
+| prefix | This is the destination prefix for IPv4 or IPv6 routes.
+| interface | Specifies the interface to be used for this route. Specify nexthop-vrf correctly if interface is in another VRF.
+| next-hop-ip | This provides the gateway IP for the prefix. Specify nexthop-vrf correctly if this IP is in another VRF.
+| distance Metric| Specifies distance value for this route. Value from 1 to 255.
+|nexthop-vrf| Specifies nexthop-vrf if interface or next-hop-ip is in another VRF.
+|blackhole| Specifies if this nexthop is to be blackholed. Only distance metric is relevant for blackhole.
+
+
+Example:
+
+#### IPv4 examples
+````
+sonic(config)# ip route 10.1.1.1/24 10.1.1.3
+
+In configDB new STATIC_ROUTE table will be filled with following entries for default VRF and prefix:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default
+prefix              = 10.1.1.1/24
+nexthop             = 10.1.1.3
+ifname              = ""
+distance            = 0
+nexthop-vrf         = ""
+blackhole           = false
+
+
+sonic(config)# ip route 10.1.1.1/24 Ethernet 12 nexthop-vrf Vrf-RED  20
+Assumption is Ethernet12 is bound to Vrf-RED.
+STATIC_ROUTE table entries are updated for newly added route:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default
+prefix              = 10.1.1.1/24
+nexthop             = 10.1.1.3, 0.0.0.0
+ifname              = "", Ethernet12
+distance            = 0, 20
+nexthop-vrf         = "", Vrf-RED
+blackhole           = false, false
+
+Note: "show ip route" will also show installed static routes along with other routes.
+
+sonic(config)# do show ip route vrf <vrf-name>
+````
+
+````
+
+sonic(config)# ip route vrf Vrf-RED 10.5.6.6/24 10.5.6.1 nexthop-vrf default 10
+
+Assumption is 10.5.6.1 is in default VRF.
+A new STATIC_ROUTE table will be filled with following entries for Vrf-RED and prefix:-
+key                 = STATIC_ROUTE|Vrf-RED|10.5.6.6/24;
+vrf-name            = Vrf-RED
+prefix              = 10.5.6.6/24
+nexthop             = 10.5.6.1
+ifname              = ""
+distance            = 10
+nexthop-vrf         = default
+blackhole           = false
+
+
+
+
+````
+#### IPv6 examples
+````
+sonic(config)# ipv6 route 2::/64 Ethernet 16
+
+sonic(config)# ipv6 route 2001:FF21:1:1::/64 18:2:1::1 100
+
+sonic(config)# ipv6 route 2111:dddd:0eee::22/128 blackhole 200
+
+Note: "show ipv6 route"  will show static routes along with other routes.
+
+sonic(config)# do show ipv6 route vrf <vrf-name>
+
+````
+##### 3.6.2.2.2 ipv4 /ipv6 Command to delete a next-hop.
+
+The vrf is an optional parameter for default instance.
+
+```
+no ip route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]|[blackhole]}

```

Comment:
It appears that the vrf and nexthop-vrf options must come after than <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]|[blackhole]}. Otherwise, it would complain for unknown command.

Reply:
We're not following VTYSH CLI command for mgmt framework CLI, this is industry standard CLI.